erpnext/asset/doctype/asset/asset.js

```diff
@@ -0,0 +1,304 @@
+			frm.page.set_inner_btn_group_as_primary(__("Make"));
+			frm.trigger("setup_chart");
+		}
+	},
+
+	setup_chart: function(frm) {
+		var x_intervals = [frm.doc.purchase_date];
+		var asset_values = [frm.doc.gross_purchase_amount];
+		var last_depreciation_date = frm.doc.purchase_date;
+
+		if(frm.doc.opening_accumulated_depreciation) {
+			last_depreciation_date = frappe.datetime.add_months(frm.doc.next_depreciation_date,
+				-1*frm.doc.frequency_of_depreciation);
+
+			x_intervals.push(last_depreciation_date);
+			asset_values.push(flt(frm.doc.gross_purchase_amount) -
+				flt(frm.doc.opening_accumulated_depreciation));
+		}
+
+		$.each(frm.doc.schedules || [], function(i, v) {
+			x_intervals.push(v.schedule_date);
+			var asset_value = flt(frm.doc.gross_purchase_amount) - flt(v.accumulated_depreciation_amount);
+			if(v.journal_entry) {
+				last_depreciation_date = v.schedule_date;
+				asset_values.push(asset_value);
+			} else {
+				if (in_list(["Scrapped", "Sold"], frm.doc.status)) {
+					asset_values.push(null);
+				} else {
+					asset_values.push(asset_value)
+				}
+			}
+		});
+
+		if(in_list(["Scrapped", "Sold"], frm.doc.status)) {
+			x_intervals.push(frm.doc.disposal_date);
+			asset_values.push(0);
+			last_depreciation_date = frm.doc.disposal_date;
+		}
+
+		frm.dashboard.render_graph({
+			title: "Asset Value",
+			data: {
+				labels: x_intervals,
+				datasets: [{
+					color: 'green',
+					values: asset_values,
+					formatted: asset_values.map(d => d.toFixed(2))
+				}]
+			},
+			type: 'line'
+		});
+	},
+
+
+	item_code: function(frm) {
+		if(frm.doc.item_code) {
+			frappe.call({
+				method: "erpnext.asset.doctype.asset.asset.get_item_details",
+				args: {
+					item_code: frm.doc.item_code
+				},
+				callback: function(r, rt) {
+					if(r.message) {
+						$.each(r.message, function(field, value) {
+							frm.set_value(field, value);
+						})
+					}
+				}
+			})
+		}
+	},
+
+	is_existing_asset: function(frm) {
+		frm.toggle_reqd("next_depreciation_date", (!frm.doc.is_existing_asset && frm.doc.calculate_depreciation));
+	},
+
+	opening_accumulated_depreciation: function(frm) {
+		erpnext.asset.set_accululated_depreciation(frm);
+	},
+
+	depreciation_method: function(frm) {
+		frm.events.make_schedules_editable(frm);
+	},
+
+	make_schedules_editable: function(frm) {
+		var is_editable = frm.doc.depreciation_method==="Manual" ? true : false;
+		frm.toggle_enable("schedules", is_editable);
+		frm.fields_dict["schedules"].grid.toggle_enable("schedule_date", is_editable);
+		frm.fields_dict["schedules"].grid.toggle_enable("depreciation_amount", is_editable);
+	},
+
+	make_purchase_invoice: function(frm) {
+		frappe.call({
+			args: {
+				"asset": frm.doc.name,
+				"item_code": frm.doc.item_code,
+				"gross_purchase_amount": frm.doc.gross_purchase_amount,
+				"company": frm.doc.company,
+				"posting_date": frm.doc.purchase_date
+			},
+			method: "erpnext.asset.doctype.asset.asset.make_purchase_invoice",
```

Comment:
please check if `erpnext.asset.doctype.asset.asset.make_purchase_invoice` can be change from `frappe.client.insert` or `frappe.client.submit`